pkg/library/libraryinputresources/resource.go

```diff
@@ -35,9 +36,19 @@ func (r Resource) ID() string {
 }
 
 func LenientResourcesFromDirRecursive(location string) ([]*Resource, error) {
+	discoveryClient, err := NewDiscoveryClientFromMustGather(location)
+	if err != nil {
+		return nil, err
+	}
+
+	_, apiResourceList, _, err := discoveryClient.GroupsAndMaybeResources()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get api resource list: %w", err)
+	}
+
```

Comment:
invert the list into `map[gvk][]gvr` here.  No need to do it for every resoruce.

Reply:
```go
mapping := map[gvk][]gvr{}
for _, apiResource := range apiResourceList.APIResources {
  gvr := schema.GroupVersionResource{
		Group:    gvk.Group,
		Version:  gvk.Version,
		Resource: apiResource.Name,
	}
  mapping[apiResource.Kind] = append(mapping[apiResource.Kind], gvr)
}

```

the pass `mapping` into `ResourceFromFile(mapping, currLocation, location)`